From f6d1bc8ecc9f50a6dd8b1f11463e91403c4b4e52 Mon Sep 17 00:00:00 2001
From: Subrata Chatterjee <schatterjee@habana.ai>
Date: Thu, 29 Jul 2021 06:56:05 +0300
Subject: [PATCH] [SW-45031] greco: avoid pldm specific duplications

In this patch, Greco HL_TARGET pldm specific
common code moved to common dir.

Change-Id: I7263f4410f97f0cce54b4e9263c813695e9bdbb6
Signed-off-by: Subrata Chatterjee <schatterjee@habana.ai>
---

diff --git a/arch/greco/arc/pldm.c b/arch/greco/arc/pldm.c
index d84dfc2..7726d76 100644
--- a/arch/greco/arc/pldm.c
+++ b/arch/greco/arc/pldm.c
@@ -48,11 +48,6 @@
 	return 0;
 }
 
-u64 hl_pll_timestamp_freq_get(void)
-{
-	return SYS_CLOCK_FREQUENCY;
-}
-
 void hl_dev_pll_init(void)
 {
 	/* Bump up UART frequency */
@@ -65,13 +60,3 @@
 	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_9);	/* UART0 TXD */
 	preboot_uart_init(UART_CLOCK_FREQUENCY / PLDM_UART_BAUDRATE);
 }
-
-u32 hl_spi_baudr_get(void)
-{
-	u32 factor = 2; /* Min Divider */
-
-	/* baud rate doesn't support odd number */
-	return (factor + 1) & 0xFFFE;
-}
-
-void preboot_init_pcie_phy(void) {}
diff --git a/arch/greco/arc_elf32/pldm.c b/arch/greco/arc_elf32/pldm.c
index d9b3282..188cf68 100644
--- a/arch/greco/arc_elf32/pldm.c
+++ b/arch/greco/arc_elf32/pldm.c
@@ -19,23 +19,7 @@
 	return 0;
 }
 
-u64 hl_pll_timestamp_freq_get(void)
-{
-	return SYS_CLOCK_FREQUENCY;
-}
-
 void hl_uart_init(void)
 {
 	preboot_uart_init(UART_CLOCK_FREQUENCY / PLDM_UART_BAUDRATE);
 }
-
-u32 hl_spi_baudr_get(void)
-{
-	u32 factor = 2; /* Min Divider */
-
-	/* baud rate doesn't support odd number */
-	return (factor + 1) & 0xFFFE;
-}
-
-void preboot_init_pcie_phy(void) {}
-
diff --git a/arch/greco/arm/pldm.c b/arch/greco/arm/pldm.c
index 4b9e506..45ef8f6 100644
--- a/arch/greco/arm/pldm.c
+++ b/arch/greco/arm/pldm.c
@@ -20,23 +20,7 @@
 	return 1;
 }
 
-u64 hl_pll_timestamp_freq_get(void)
-{
-	return SYS_CLOCK_FREQUENCY;
-}
-
 void hl_uart_init(void)
 {
 	preboot_uart_init(UART_CLOCK_FREQUENCY / PLDM_UART_BAUDRATE);
 }
-
-u32 hl_spi_baudr_get(void)
-{
-	u32 factor = 2; /* Min Divider */
-
-	/* baud rate doesn't support odd number */
-	return (factor + 1) & 0xFFFE;
-}
-
-void preboot_init_pcie_phy(void) {}
-
diff --git a/arch/greco/common/pldm.c b/arch/greco/common/pldm.c
index e7fd49f..d692bd4 100644
--- a/arch/greco/common/pldm.c
+++ b/arch/greco/common/pldm.c
@@ -5,4 +5,21 @@
  *
  */
 
-/* TODO: Greco pldm(HL_TARGET) specific common functions to be moved here */
+#include <preboot.h>
+
+#define SYS_CLOCK_FREQUENCY	155000ULL
+
+u64 hl_pll_timestamp_freq_get(void)
+{
+	return SYS_CLOCK_FREQUENCY;
+}
+
+u32 hl_spi_baudr_get(void)
+{
+	u32 factor = 2; /* Min Divider */
+
+	/* baud rate doesn't support odd number */
+	return (factor + 1) & 0xFFFE;
+}
+
+void preboot_init_pcie_phy(void) {}
